cli/flags.rs

```diff
@@ -56,106 +74,114 @@ pub struct DenoFlags {
-        .long("allow-env")
-        .help("Allow environment access"),
-    )
-    .arg(
-      Arg::with_name("allow-run")
-        .long("allow-run")
-        .help("Allow running subprocesses"),
-    )
-    .arg(
-      Arg::with_name("allow-hrtime")
-        .long("allow-hrtime")
-        .help("Allow high resolution time measurement"),
-    )
-    .arg(
-      Arg::with_name("allow-all")
-        .short("A")
-        .long("allow-all")
-        .help("Allow all permissions"),
-    )
-    .arg(
-      Arg::with_name("no-fetch")
-        .long("no-fetch")
-        .help("Do not download remote modules"),
-    )
-}
+    DENO_DIR       Set deno's base directory
+    NO_COLOR       Set to disable color
+    HTTP_PROXY     Proxy address for HTTP requests (module downloads, fetch)
+    HTTPS_PROXY    Same but for HTTPS";
 
-pub fn create_cli_app<'a, 'b>() -> App<'a, 'b> {
-  add_run_args(App::new("deno"))
-    .bin_name("deno")
-    .global_settings(&[AppSettings::ColorNever, AppSettings::UnifiedHelpMessage, AppSettings::DisableVersion])
-    .settings(&[AppSettings::AllowExternalSubcommands])
-    .after_help(ENV_VARIABLES_HELP)
-    .long_about("A secure JavaScript and TypeScript runtime
+static DENO_HELP: &str = "A secure JavaScript and TypeScript runtime
 
-Docs: https://deno.land/manual.html
+Docs: https://deno.land/std/manual.md
 Modules: https://deno.land/x/
 Bugs: https://github.com/denoland/deno/issues
 
-To run the REPL:
+To run the REPL supply no arguments:
 
   deno
 
-To execute a sandboxed script:
+To evaluate code from the command line:
+
+  deno eval \"console.log(30933 + 404)\"
+
+To execute a script:
 
   deno https://deno.land/std/examples/welcome.ts
 
-To evaluate code from the command line:
+The default subcommand is 'run'. The above is equivalent to
 
-  deno eval \"console.log(30933 + 404)\"
+  deno run https://deno.land/std/examples/welcome.ts
 
-To get help on the another subcommands (run in this case):
+See 'deno help run' for run specific flags.";
 
-  deno help run")
-    .arg(
-      Arg::with_name("version")
-        .short("v")
-        .long("version")
-        .help("Print the version"),
-    )
+/// Main entry point for parsing deno's command line flags.
+/// Exits the process on error.
+pub fn flags_from_vec(args: Vec<String>) -> DenoFlags {
+  match flags_from_vec_safe(args) {
+    Ok(flags) => flags,
+    Err(err) => err.exit(),
+  }
+}
+
+/// Same as flags_from_vec but does not exit on error.
+pub fn flags_from_vec_safe(args: Vec<String>) -> clap::Result<DenoFlags> {
+  let args0 = args[0].clone();
+  let args = arg_hacks(args);
+  let app = clap_root();
+  let matches = app.get_matches_from_safe(args)?;
+
+  let mut flags = DenoFlags::default();
+
+  flags.argv.push(args0);
+
+  if matches.is_present("log-level") {
+    flags.log_level = match matches.value_of("log-level").unwrap() {
+      "debug" => Some(Level::Debug),
+      "info" => Some(Level::Info),
+      _ => unreachable!(),
```

Comment:
Something better than "unreachable!" here? `deno --log-level warn` now begets this nasty crash.

Reply:
nvm, doesn't actually seem to happen